File: java-workshop/part-4-behavior/src/main/java/service/DiscountCodeService.java
package service;

import domain.DiscountCode;
import repository.DiscountCodeRepository;

/**
 * Discount code validation service.
 */
public class DiscountCodeService {
    private final DiscountCodeRepository discountCodeRepository;

    public DiscountCodeService(DiscountCodeRepository discountCodeRepository) {
        this.discountCodeRepository = discountCodeRepository;
    }

    /**
     * Checks a discount code for a given customer.
     *
     * @return true if the code has already been used by this customer
     */
    public boolean checkDiscountCode(String customerId, DiscountCode discountCode) {
        return discountCodeRepository.checkDiscountCode(customerId, discountCode);
    }

    public void markAsUsed(String customerId, DiscountCode discountCode) {
        discountCodeRepository.markAsUsed(customerId, discountCode);
    }
}

File: csharp-workshop/Part2.LightMocks/Service/DiscountCodeService.cs
using Part2.LightMocks.Domain;
using Part2.LightMocks.Repository;

namespace Part2.LightMocks.Service;

/// <summary>
/// Discount code validation service.
/// </summary>
public class DiscountCodeService
{
    private readonly IDiscountCodeRepository _discountCodeRepository;

    public DiscountCodeService(IDiscountCodeRepository discountCodeRepository)
    {
        _discountCodeRepository = discountCodeRepository;
    }

    /// <summary>
    /// Checks a discount code for a given customer.
    /// </summary>
    /// <returns>true if the code has already been used by this customer</returns>
    public bool CheckDiscountCode(string customerId, DiscountCode discountCode)
    {
        return _discountCodeRepository.CheckDiscountCode(customerId, discountCode);
    }

    public void MarkAsUsed(string customerId, DiscountCode discountCode)
    {
        _discountCodeRepository.MarkAsUsed(customerId, discountCode);
    }
}
